### ISSUE TITLE
Add styling for program cards

### ISSUE THREAD

**Issue by ssandbekkhaug** [civiform/civiform#8157]
![Image](https://github.com/user-attachments/assets/2d6046a5-17b8-453d-b1cd-05fc19eabe6b)

[Program card component mocks](https://www.figma.com/design/RmhVJCLgAKWGENJLWWAtNS/CiviForm%3A-Components%2C-Styles?node-id=3764-4671&t=Fa1waxVlaJPWYACD-1)

We will be using the [USWDS cards](https://designsystem.digital.gov/components/card/).

Make sure to check screen reader behavior.  The screen reader should start with the heading of each card, then read the image alt text and the categories.  Also, consider if we want to include a SR label for each part of the card, so that it's clear what's being read.  For example:  "program description", or "program categories", etc.

The screen-reader-only text for the program card "View and apply" buttons should be "View and apply to Program name". 

### Color guidelines
- CiviForm and USWDS colors: Unless specified all North Star designs will be implemented using [USWDS theme color tokens](https://designsystem.digital.gov/design-tokens/color/theme-tokens/) and [USWDS state color tokens](https://designsystem.digital.gov/design-tokens/color/state-tokens/) 

### Text styles
- All text should use the class `usa-prose` unless specified otherwise 
- https://designsystem.digital.gov/components/prose/

### Hover states
- Use the default USWDS hover styles.

### USWDS components used
- [USWDS tag](https://designsystem.digital.gov/components/tag/)
    - See USWDS guidance on the how to use the component
    - Our version will be using sentence case 
    - See Figma for detailed color specifications
- [USWDS card](https://designsystem.digital.gov/components/card/)
    - See USWDS guidance on the how to use the component
    - Our version has the addition of tags above and below the main content
    - We have versions with images and without images(for my applications section)
    - See Figma for detailed color specifications

**Comment by melanie-exygy** [comment]
@sijiayam I think you said that mocks for this were finalized.  Would you mind adding a link to them here and a screenshot?

**Comment by sijiayam** [comment]
Design QA complete. Issue opened: #9412
